codec/decode.go

```diff
@@ -734,6 +734,15 @@ func (f *decFnInfo) kStruct(rv reflect.Value) {
 					} else {
 						d.decodeValue(si.field(rv, true), nil)
 					}
+				} else if fti.ufh {
```

Comment:
A fully general solution would also implement the else branch below (for formats that don't have length-prefixed maps, I guess)